core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowExecutionsResource.scala

```diff
@@ -131,6 +157,24 @@ class WorkflowExecutionsResource {
       .execute();
   }
 
+  /** Delete a group of executions */
+  @PUT
+  @Path("/delete_executions")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  def groupDeleteExecutionsOfWorkflow(
+      request: ExecutionGroupDeleteRequest,
+      @Auth sessionUser: SessionUser
+  ): Unit = {
+    validateUserCanAccessWorkflow(sessionUser.getUser.getUid, request.wid)
+    /* delete the execution in sql */
+    for (eId <- request.eIds) {
```

Comment:
we are trying to do the group deletion in the backend to avoid the loop. Can you group delete using the SQL statement?